### PR TITLE
Ensure that feeds are appropriately restricted (#10018)

### LINKED ISSUE #9981: Private repo under User activities must hidden!!!
- Gitea version (or commit ref):
- Git version: 2.25
- Operating system: ubuntu 28.04
- Database (use `[x]`):
  - [ ] PostgreSQL
  - [x]MySQL
  - [ ] MSSQL
  - [ ] SQLite
- Can you reproduce the bug at https://try.gitea.io:
  - [ ] Yes (provide example URL)
  - [ ] No
  - [x] Not relevant
- Log gist:

## Description

Visitor (dont autenticated) can view user activities about a private repo. This is mistake. Visitors will never recognize the existence of a private repo.

### PR DESCRIPTION
**Pull request by zeripath**
Backport #10018 - Not a direct backport as that area of code has been changed with the restricted user implementation

Fix #9981

### CODE REVIEW

**Review by guillep2k — APPROVED** [review]

**Comment by guillep2k** [comment]
Just make golangci-lint happy. 😁

File: routers/user/profile.go [hunk 1]
```diff
@@ -156,14 +156,20 @@ func Profile(ctx *context.Context) {
 		orderBy = models.SearchOrderByRecentUpdated
 	}
 
+	requestingUserId := int64(0)
```
Comment (zeripath) [inline routers/user/profile.go]:
```suggestion
	requestingUserID := int64(0)
```

File: routers/user/profile.go [hunk 2]
```diff
@@ -156,14 +156,20 @@ func Profile(ctx *context.Context) {
...
+	if ctx.User != nil {
+		requestingUserId = ctx.User.ID
```
Comment (zeripath) [inline routers/user/profile.go]:
```suggestion
		requestingUserID = ctx.User.ID
```

File: routers/user/profile.go [hunk 3]
```diff
@@ -156,14 +156,20 @@ func Profile(ctx *context.Context) {
...
+
 	keyword := strings.Trim(ctx.Query("q"), " ")
 	ctx.Data["Keyword"] = keyword
 	switch tab {
 	case "activity":
 		retrieveFeeds(ctx, models.GetFeedsOptions{RequestedUser: ctxUser,
-			IncludePrivate:  showPrivate,
-			OnlyPerformedBy: true,
-			IncludeDeleted:  false,
+			RequestingUserID: requestingUserId,
```
Comment (zeripath) [inline routers/user/profile.go]:
```suggestion
			RequestingUserID: requestingUserID,
```

**Comment by zeripath** [comment]
Golangci-lint argh!!

**Review by mrsdizzie — APPROVED** [review]

**Review by lafriks — APPROVED** [review]

**Pull request merged**